NGitLab.Mock/Repository.cs

```diff
@@ -473,6 +475,22 @@ internal IEnumerable<Models.Tree> GetTree(Models.RepositoryGetTreeOptions reposi
             }
         }
 
+        internal string FetchBranchFromFork(Project fork, string branch)
+        {
+            if (fork?.ForkedFrom != Project)
+                throw new InvalidOperationException("Cannot fetch MR source branch from unrelated project");
+
+            var repo = GetGitRepository();
+            var forkRemoteName = "fork"; // Should we also include the name of the fork?
```

Comment:
I think it should be unique per fork. If you have multiple operation in parallel, this is not safe.

Clients are safe, because all code run in a lock. However, you can manipulate the mock objects directly, and there is no thread-safe guarantee.

```suggestion
            var forkRemoteName = "fork" + fork.ForkedFrom.Id;
```